### DISCUSSION TITLE
Best Settings for a 4090?

### DISCUSSION THREAD

**Discussion by gohan2091** [vladmandic/sdnext#2418] [Q&A]
I've been on Google and found a few posts but these are months old so likely out of date. I would like to know specifically for the SD.Next fork, what settings should I have enabled (or disabled) to get the best out of my 4090?

Full precision for model (--no-half) and Full precision for VAE (--no-half-vae) I have enabled because for example ControlNet Reference mode throws an error if I have these disabled. "(ControlNet failed to use VAE. Please try to add `--no-half-vae`, `--no-half` and remove `--precision full` in launch cmd.)" I understand these two settings slow performance. In tests, running batch counts higher than 1 or using hi-res fix, the speed drops by over half with these both enabled. Do I need them?

For device precision type I have BF16 enabled. I have Upscast sampling and upcast cross attention layer and fixed UNet precision disabled. Should these be enabled? Xformers are disabled but Scaled-Dot-Product is enabled. There are a load of settings in the Diffuser Settings too which I am not sure about. Any help from people who know what they're talking about would be appreciated. I have included a screenshot of my System Info. It shows CUDA 12.1 but I actually have 12.3 installed. My CPU is an AMD Ryzen 9 5950X. Not interested in tensorRT at the moment, seems a lot of hassle and with some limitations. 

![Screenshot 2023-10-27 211134](https://github.com/vladmandic/automatic/assets/127342824/6b2ffb61-76d0-447a-ba97-80811c3b5c7f)

**Comment by vladmandic** [comment]
you have cuda 12.3 installed and only possibly causing conflicts. torch comes with cuda prepackaged - and sdnext default torch is 2.1.0 with cuda 12.1.

otherwise, you're asking for ideal settings, but you're upcasting or forcing fp32 in many places. if you want ideal, get a way to get controlnet reference working without fp32, use fp16 vae and do not upcast anything.

**Reply by gohan2091** [reply]
Ah I didn't know about CUDA coming prepackaged. I have turned off Full precision for model (--no-half) and Full precision for VAE (--no-half-vae). I also enabled hypertile. I have gained a big speed boost, especially with hires.fix. I don't know how to fix the ControlNet Reference error message I posted above. If anyone has any ideas, please let me know. To lose so much speed just for ControlNet Reference is not worth it.
